tests/Text/MMarkSpec.hs

```diff
@@ -382,10 +383,18 @@ spec = parallel $ do
     context "4.7 Link reference definitions" $ do
       it "CM159" $
         "[foo]: /url \"title\"\n\n[foo]" ==->
+#ifdef ghcjs_HOST_OS
+          "<p><a title=\"title\" href=\"/url\">foo</a></p>\n"
```

Comment:
This seems to be unrelated to the actual change in this PR. Do you have any idea why the order of attributes is different with GHCJS?

Reply:
Lucid is using a hashmap, which is unordered by design. I presume that JS is using a different hashing function. Having lucid use an ordered map instead of a unordered one would resolve the difference.